sys-kernel/dracut/dracut-9999.ebuild

```
@@ -22,7 +22,8 @@ HOMEPAGE="https://github.com/dracut-ng/dracut-ng/wiki"
 LICENSE="GPL-2"
 SLOT="0"
 IUSE="selinux test"
-RESTRICT="!test? ( test )"
+RESTRICT="test"
+PROPERTIES="test_privileged test_network"
```

I think this should technically still be behind a USE-conditional, but I'm not sure if Portage evaluates it properly in PROPERTIES.
```suggestion
RESTRICT="!test( test) "
PROPERTIES="test? ( test_privileged test_network )"
```

---

That would unrestrict the test feature if the test flag is enabled. It is my understanding that when using these properties the tests should be restricted always, and then effectively `ALLOW_TEST` overrides this `RESTRICT` based on the `PROPERTIES`.

---

It should, at least PMS specifies that USE conditionals are legal there.

It should be `RESTRICT="!test? ( test )"` however.

---

Nevermind on this for now; I think Portage needs a code change to handle this properly. Your original change should suffice.

---

Based on the PMS description of test_privileged:

> The package manager may run tests that require superuser privileges, even if the ebuild has RESTRICT=test.

I suppose it would be correct to put a USE-conditional in PROPERTIES, but leave RESTRICT=test unconditional.
But again, that would require testing to ensure Portage handles USE conditions in PROPERTIES correctly.

Really, I just want to ensure that the test phase cannot be run with the "test" USE flag disabled, since that would lead to unsatisfied dependencies.

---

I did a quick test, and this seems to work properly.
```suggestion
RESTRICT="test"
PROPERTIES="test? ( test_privileged test_network )"
```

With FEATURES="test" USE="test" ALLOW_TEST="network privileged" the test phase is executed as expected.

With FEATURES="test" USE="-test" ALLOW_TEST="network privileged" the test phase is skipped.

---

Fixed, Thanks for testing!